File: since2020/ZeroJudge/ZeroJudge a223.py
'''
	@judge ZeroJudge
	@id a233
	@name Power Strings
	@source UVa 10298

	@tag String Manipulation, Integer Factorisation, Search
'''
from sys import stdin
from math import ceil, sqrt
from itertools import chain

def test(s, x):
	n = len(s)
	t = s[:x] * (n // x)
	return s == t

def solve(s):
	n = len(s)
	divisor = [ x for x in range(1, ceil(sqrt(n)) + 1) if n % x == 0 ]
	for d in chain(divisor, map(lambda x: n // x, reversed(divisor))):
		if test(s, d):
			return n // d
	return 1

for line in stdin:
	if line.strip() != '.':
		print(solve(line.strip()))

File: since2020/ZeroJudge/ZeroJudge a223.cpp
/**
 *  @judge ZeroJudge
 *  @id a233
 *  @name Power Strings
 *  @source UVa 10298
 * 
 *  @tag String Manipulation, Integer Factorisation, Search
 */
#include <bits/stdc++.h>
using namespace std;

string s;

bool test(int x) {
	for (int i = x; i < s.size(); i += x)
		for (int j = 0; j < x; j++)
			if (s[j] != s[i + j])
				return false;
	return true;
}

int solve() {
	int n = sqrt((double) s.size() + 0.001);
	vector<int> divisor;
	for (int i = 1; i <= n; i++) {
		if (s.size() % i == 0) {
			divisor.push_back(i);
			divisor.push_back(s.size() / i);
		}
	}
	sort(divisor.begin(), divisor.end());
	for (const auto &i : divisor)
		if (test(i))
			return s.size() / i;
	return 1;
}

int main() {
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	cout.tie(nullptr);
	while (cin >> s, s != ".")
		cout << solve() << endl;
}